x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -661,9 +660,8 @@ protected void doSaveState(IndexerState indexerState, DataFrameIndexerPosition p
 
             // Persisting stats when we call `doSaveState` should be ok as we only call it on a state transition and
```

Comment:
nit: this comment looks outdated, quick suggestion:

"Persist the current state and stats in the internal index. The interval of this method being called is controlled by AsyncTwoPhaseIndexer#onBulkResponse which calls doSaveState every-so-often when doing bulk indexing calls or at the end of one indexing run"